deltachat-ios/Controller/WebxdcViewController.swift

```diff
@@ -62,24 +62,37 @@ class WebxdcViewController: WebViewViewController {
         
         let script = """
         window.webxdc = (() => {
-          let setUpdateListenerPromise = null
           var log = (s)=>webkit.messageHandlers.log.postMessage(s);
         
           var update_listener = () => {};
 
-          window.__webxdcUpdate = async (lastSerial) => {
+          let should_run_again = false;
+          let running = false;
+          let lastSerial = 0;
+          window.__webxdcUpdate = async () => {
+            if (running) {
+                should_run_again = true
+                return
+            }
+            if (should_run_again) {
+                should_run_again = false
+            }
+            running = true;
             try {
-                const updates = await fetch("webxdc-update.json?"+lastSerial).then((response) => response.json())
+                var updates = await fetch("webxdc-update.json?"+lastSerial).then((response) => response.json())
                 updates.forEach((update) => {
                   update_listener(update);
+                  if (lastSerial < update["max_serial"]){
+                    lastSerial = update["max_serial"]
+                  }
                 });
             } catch (e) {
                 log("json error: "+ e.message)
             } finally {
-              if (setUpdateListenerPromise) {
-                 setUpdateListenerPromise()
-                 setUpdateListenerPromise = null
-              }
+                running = false;
+                if (should_run_again) {
+                    window.__webxdcUpdate()
```

Comment:
> i am also not sure, that they are correct in this form: the doc of [setUpdateListener](https://docs.webxdc.org/spec.html#setupdatelistener) says "The returned promise resolves when the listener has processed all the update messages known at the time when setUpdateListener was called." - now, there could be more updates processed if during processing new updates come in.

should be fixed by adding a second await here?
```suggestion
                    await window.__webxdcUpdate()
```

Reply:
can you elaborate why the `await` solves the issue?

Reply:
I'm sorry, but I don't really see any problem here.

Reply:
i am not saying that there are problems.

i just want to understand - imu, this await will wait until the promise is resolved and then return the promise from the outer `__webxdcUpdate()` call - this seems wronger than without the `await` as there could be more updates processed if during processing new updates come in.

what happens if the `await` was left out as before? would the function be scheduled at some later point? that would probably what we want to have here.

we can also do an av on that.